b2luigi/batch/processes/gbasf2.py

```diff
@@ -0,0 +1,764 @@
+        # gbasf2 job type (e.g. User, Production, ...)
+        jobtype = get_setting("gbasf2_jobtype", default=False, task=self.task)
+        if jobtype is not False:
+            gbasf2_command_str += f" --jobtype {jobtype} "
+
+        # additional basf2 options to use on grid
+        basf2opt = get_setting("gbasf2_basf2opt", default=False, task=self.task)
+        if basf2opt is not False:
+            gbasf2_command_str += f" --basf2opt='{basf2opt}' "
+
+        # optional string of additional parameters to append to gbasf2 command
+        gbasf2_additional_params = get_setting("gbasf2_additional_params", default=False, task=self.task)
+        if basf2opt is not False:
+            gbasf2_command_str += f" {gbasf2_additional_params} "
+
+        gbasf2_command = shlex.split(gbasf2_command_str)
+        return gbasf2_command
+
+    def _write_path_to_file(self):
+        """
+        Serialize and save the ``basf2.Path`` returned by ``self.task.create_path()`` to a python pickle file.
+        """
+        try:
+            path = self.task.create_path()
+        except AttributeError as err:
+            raise Exception(
+                "Gbasf2 batch process can only used with tasks that generate basf2 paths with "
+                "a ``create_path()`` method, e.g. are an instance of ``Basf2PathTask``."
+            ) from err
+
+        path.add_module("Progress")
+        b2pp.write_path_to_file(path, self.pickle_file_path)
+
+    def _create_wrapper_steering_file(self):
+        """
+        Create a steering file to send to the grid that executes the pickled
+        basf2 path from ``self.task.create_path()``.
+        """
+        # read a jinja2 template for the steerinfile that should execute the pickled path
+        template_file_path = os.path.join(self._file_dir, "templates/gbasf2_steering_file_wrapper.jinja2")
+        with open(template_file_path, "r") as template_file:
+            template = Template(template_file.read())
+            # replace some variable values in the template
+            steering_file_stream = template.stream(
+                pickle_file_path=self.pickle_file_path,
+                max_event=get_setting("max_event", default=0, task=self.task),
+            )
+            # write the template with the replacements to a new file which should be sent to the grid
+            steering_file_stream.dump(self.wrapper_file_path)
+
+    def _download_dataset(self):
+        """
+        Download the results from a gbasf2 project, stored as a dataset on the grid.
+        """
+        # Get list of files that we want to download from the grid via ``gb2_ds_list`` so that we can
+        # then compare this list with the results of the download to see if it was successful
+        ds_list_command = shlex.split(f"gb2_ds_list --user {self.dirac_user} {self.gbasf2_project_name}")
+        output_dataset_str = subprocess.run(ds_list_command, check=True, env=self.gbasf2_env, stdout=PIPE, encoding="utf-8").stdout
```

Comment:
I used `subprocess.run(..., check=True, stdout=PIPE).stdout instead of `subprocess.check_output` in most cases, because the current subprocess documentation recommends to use the universal `run` high level interface. However, it's much more verbose in python 3.6, starting from 3.7 I could have just set the new `capture_output` argument of `run` to `True`.

Reply:
I added a `run_with_gbasf2` which runs `subprocess.run` with the gbasf2 environment and changed default settings. Also I added the `capture_output` argument from python 3.8.